### ISSUE TITLE
alternator refactoring open questions

### ISSUE THREAD

**Issue by rusefillc** [rusefi/rusefi#5716]
https://github.com/rusefi/rusefi/commit/3e9b2d0e1f87f4b6990e4a16f81b1b6c9ca4e2a7#r132373139 and https://github.com/rusefi/rusefi/commit/3e9b2d0e1f87f4b6990e4a16f81b1b6c9ca4e2a7#r132373388 should be addressed

**Comment by andreika-git** [comment]
We are passing parameter by reference, so until we start using PID, everything is fine. We have time to read the config and fill in the PID settings before we start using it.

**Comment by mck1117** [comment]
> We are passing parameter by reference, so until we start using PID, everything is fine. We have time to read the config and fill in the PID settings before we start using it.

That doesn't mean it isn't UB. You can't guarantee the behavior of the `->` operator until the object has been fully initialized.

**Comment by mck1117** [comment]
In fact - the current implementation even relies on the static constructor order. It's not required that `engineConfiguration` be optimized out by the compiler, which means it may be null until all static construction is complete. It so happens that the compiler _does_ make that optimization, but it's not a guarantee.

**Comment by andreika-git** [comment]
> > We are passing parameter by reference, so until we start using PID, everything is fine. We have time to read the config and fill in the PID settings before we start using it.
> 
> That doesn't mean it isn't UB. You can't guarantee the behavior of the `->` operator until the object has been fully initialized.

True, that's not guaranteed. The problem existed before the refactoring. We should either move it to init() called from initAlternatorCtrl(), or pass &persistentState.persistentConfiguration (but there's also EFI_UNIT_TEST so it's more complicated).

**Comment by andreika-git** [comment]
https://github.com/rusefi/rusefi/pull/5717
